### ISSUE TITLE
Angular tutorial - cannot find name HEROES. and also one other pedantic point

### ISSUE THREAD

**Issue by carlosthehyena** [angular/angular#18552]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Feature request
[X ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

Parrt 3 of tutorial - MasterDetail.
https://angular.io/tutorial/toh-pt2
heroes = HEROES;
Browser error: Cannot find name 'HEROES'.
Studio code warning: [ts] Cannot find name HEROES as a warning
If I replace with heroes:Hero[] = HEROES, it resolves.

On another minor ( but nonetheless important point)

Part 2 if tutorial: The Hero Editor
https://angular.io/tutorial/toh-pt1

Quote-
Add two properties to the AppComponent: a title property for the app name and a hero property for a hero named "Windstorm."

At this point, the title property already exists. Adding of course will result in "Duplicate Identifier warning". Pendantic I know, but good to get a tutorlal to follow without errors for us newbsters.

## Environment

<pre><code>
Angular version: 4.0.0

Browser:
- [ ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by gkalpak** [comment]
As you can see in the [live example][1], `heroes = HEROES` works without errors. 
Please give exact steps to reproduce the issue. It sounds more like you missed a previous step or something.

Regarding your second point, there shouldn't be a `title` property on `AppComponent` already.
It is the very first code change instructed in the tutorial, so at this point `AppComponent` should have no `title` property 😕 

[1]: https://angular.io/generated/live-examples/toh-pt2/eplnkr.html

**Comment by gkalpak** [comment]
Closing since we didn't get any feedback and everything should work as expected afaict.

**Comment by andrew-lynch** [comment]
I had the same error where `heroes = HEROES` could not be found. However, once I restarted the server everything worked correctly.

**Comment by hithto** [comment]
+1

toh-pt2

ERROR in src/app/app.component.ts(29,11): error TS2304: Cannot find name 'HEROES'.
webpack: Failed to compile.

**Comment by jsaintyv** [comment]
I had the same error.
But when i restarted my server. It doesn't correct.

I replace HEROES constant by function :
export function getHeroes():Hero[] {
  return [
    { id: 11, name: 'Mr. Nice' },
    { id: 12, name: 'Narco' },
    { id: 13, name: 'Bombasto' },
    { id: 14, name: 'Celeritas' },
    { id: 15, name: 'Magneta' },
    { id: 16, name: 'RubberMan' },
    { id: 17, name: 'Dynama' },
    { id: 18, name: 'Dr IQ' },
    { id: 19, name: 'Magma' },
    { id: 20, name: 'Tornado' }
  ];
}

ngOnInit() {
    this.heroes = getHeroes();
}

**Comment by ccsjesse** [comment]
I got the same error but because I wrote: 'heroes **:** HEROES;'.
Error disappears when I corrected to 'heroes **=** HEROES;'.
Found the solution from: https://translate.google.ca/translate?hl=en&sl=ru&u=https://toster.ru/q/478059&prev=search

**Comment by ashikjs** [comment]
Same problems fro me but I was miss to call it.

`import { HEROES } from './mock-heroes';`

**Comment by kaycebasques** [comment]
I got this error too. My mistake was here:

![image](https://user-images.githubusercontent.com/4713486/46185280-206da000-c28e-11e8-88b9-8709a675c323.png)

I used `heroes: HEROES` by mistake. Changing it to what the doc says fixed it.

This is probably more common with TypeScript beginners, like myself. The issue was that I wrote out the code by hand, using the code below it (`hero: HERO = {`) as my guide. If you only look at the beginning of that next line (`hero: HERO`) you might make the mistaken assumption that you use colons to define properties in TS.
